clang/lib/Sema/SemaChecking.cpp

```diff
@@ -8899,18 +8899,42 @@ void Sema::CheckMemaccessArguments(const CallExpr *Call,
           << ArgIdx << FnName << PointeeTy
           << Call->getCallee()->getSourceRange());
     else if (const auto *RT = PointeeTy->getAs<RecordType>()) {
+
+      auto IsTriviallyCopyableCXXRecord = [](auto const *RT) {
+        auto const *D = RT->getDecl();
```

Comment:
```suggestion
        const Decl *D = RT->getDecl();
```
Please only use `auto` when the type is spelled out in the initialization. Also, we use `const Type` and not `Type const` as the prevailing style, so you should stick with that.